tests/test_openapi.py

```diff
@@ -45,3 +52,339 @@ def test_str2bool():
+    Get collections from pygeoapi configuration
+    :param config: pygeoapi configuration dict
+    :returns: collections dict
+    """
+    return filter_dict_by_key_value(config['resources'],
+                                    'type',
+                                    'collection')
+
+
+def items_path(coll_name):
+    """
+    Generate items path from collection name
+    :param coll_name: collection name
+    :returns: items path
+    """
+    return '/collections/{}/items'.format(coll_name)
+
+
+def item_path(coll_name):
+    """
+    Generate item path from collection name
+    :param coll_name: collection name
+    :returns: item path
+    """
+    return '/collections/{}/items/{{featureId}}'.format(coll_name)
+
+
+def supports_transactions(collection):
+    """
+    Check if given collection supports transactions
+    :param collection: collection dict
+    :returns: boolean value
+    """
+    if 'transactions' not in collection['extents']:
+        return False
+    return collection['extents']['transactions']
+
+
+def test_name_value_pair_obj(schemas):
+    """
+    Assertions for nameValuePairObj in OpenAPI document
+    :param schemas: schemas dict
+    """
+    assert 'nameValuePairObj' in schemas
+    assert 'name' in schemas['nameValuePairObj']['properties']
+    assert 'value' in schemas['nameValuePairObj']['properties']
+
+
+def post_schema_assertions(verbs):
+    """
+    Assertions for post schema in OpenAPI document
+    :param verbs: verbs dict
+    """
+    # assertion for post
+    assert 'post' in verbs
+    post = verbs['post']
+    assert isinstance(post, dict)
+
+    # assertion for post attributes
+    post_attrib = ['summary',
+                   'description',
+                   'tags',
+                   'requestBody',
+                   'responses']
+    for attrib in post_attrib:
+        assert attrib in post
+
+    # assertion for post request attributes
+    post_req_attrib = ['required',
+                       'content']
+    post_req = post['requestBody']
+    assert isinstance(post_req, dict)
+    for attrib in post_req_attrib:
+        assert attrib in post_req
+
+    # assertion for post request content attributes
+    post_req_content_attrib = ['type',
+                               'geometry',
+                               'properties']
+    post_req_content = post_req['content']['application/geo+json']
+    assert isinstance(post_req_content, dict)
+    post_req_content_schema = post_req_content['schema']
+    assert isinstance(post_req_content_schema, dict)
+    for attrib in post_req_content_attrib:
+        assert attrib in post_req_content_schema['properties']
+
+    # assertion for post response attributes
+    post_resp_attrib = ['201',
+                        '400',
+                        '404',
+                        '500']
+    post_resp = post['responses']
+    for attrib in post_resp_attrib:
+        assert attrib in post_resp
+
+
+def patch_schema_assertions(verbs):
```

Comment:
Is this a test or a utility function? The same comment is valid for several functions below

Reply:
I have created functions to group assertions related to each schema. The test function will call them if transactions are enabled for a resource.

Reply:
I used to have assertions within a test function but I don't know if it is common this coding pattern. cc @jorgejesus

Reply:
@alex-mathew  @francbartoli 

IMHO: 
- Better more asserts than less
- `assert` inside auxiliary functions maybe is less positive but  we have easier to read code with smaller reusable functions (find this more important).
- `assert` in main code is an anti-padron and should not be used

Reply:
@jorgejesus should have anyway the label `test_*` if they are asserting something?

Reply:
@francbartoli  IMHO `test_*` from the actual test being performed, then generic function like `check_transaction_schemas_abscent` can have `assert` but they are related to the test they are being called `test_transaction_a_b` what we have now should be ok (my 2 cent)

Reply:
@jorgejesus the only concern is that those functions get out of pytest which is a little bit weird